torch/ao/quantization/experimental/observer.py

```diff
@@ -4,30 +4,112 @@
 """
 
 import torch
+import itertools
 from torch.ao.quantization.observer import ObserverBase
 
 # TODO: Consider adding NonUniformQuantizationObserverBase class
 # when more than one non-uniform method is implemented
 
 class APoTObserver(ObserverBase):
-    alpha = 0
-    gamma = 0
-    level_indices = torch.Tensor()
+    b: int
+    k: int
+    n: int
+    alpha: float
+    gamma: float
+    max_val: float
+    level_indices: torch.Tensor
 
     def __init__(
         self,
-        min_val: torch.Tensor,
-        max_val: torch.Tensor,
-        level_indices: torch.Tensor,
-            b: int,
-            k: int) -> None:
-        super().__init__
+        max_val,
+        b,
+        k,
+            dtype=torch.quint8) -> None:
+        super().__init__(dtype)
+        self.max_val = max_val
+        self.b = b
+        self.k = k
 
-    def calculate_qparams(self):
-        return self._calculate_qparams()
+    def calculate_qparams(self, signed):
+        return self._calculate_qparams(signed)
 
-    def _calculate_qparams(self):
-        raise NotImplementedError
+    r""" Calculates nonuniform quantization parameters according to APoT paper:
+    https://arxiv.org/pdf/1909.13144.pdf.
+    Arg:
+        signed: specifies whether to include signed values in quantization level calculations
+    Returns:
+        gamma: gamma quantization parameter, defined to ensure that alpha is the maximum of the range
+        quantization_levels: non-uniform quantization levels (fp representation)
+        level_indices: int representation of quantization_levels indices
+    """
+    def _calculate_qparams(self, signed):
+        # compute alpha
+        self.alpha = self.max_val
+
+        # check for valid inputs of b, k
+        assert(self.k and self.k != 0)
+        assert(self.b % self.k == 0)
+
+        # compute n and store as member variable
+        self.n = self.b // self.k
```

Comment:
nit and opinionated: whenever I see `//` in python, my mind gravitates the possibility that the denominator may not be divisible by the numerator (otherwise, why would we need `//` instead of `/`). because, by definition, APoT requires b % k == 0, it might mitigate overthinking on a reader's part if `/` was used here

Reply:
i left it as `//` here because when i change to `/` i get a TypeError that n could be a float (but was defined as an int above). since i am asserting that b % k == 0 right above, hopefully this is fine?